code/modules/power/generators/portable generators/port_gen.dm

```diff
@@ -0,0 +1,86 @@
+
+
+//Baseline portable generator. Has all the default handling. Not intended to be used on it's own (since it generates unlimited power).
+/obj/machinery/power/port_gen
+	name = "Placeholder Generator"	//seriously, don't use this. It can't be anchored without VV magic.
+	desc = "A portable generator for emergency backup power."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "portgen0_0"
+	density = TRUE
+	anchored = FALSE
+
+	var/active = FALSE
+	var/power_gen = 5000
+	var/power_output = 1
+	var/base_icon = "portgen0"
+
+/obj/machinery/power/port_gen/attack_hand(mob/user as mob)
+	if(..())
+		return
+	if(!anchored)
+		return
+
```

Comment:
Does this attack_hand here actually do anything?

Reply:
no, I removed it